src/pages/SupplierRequisitionsPage.js

```diff
@@ -38,6 +41,17 @@ export class SupplierRequisitionsPage extends React.Component {
     };
   }
 
+  // Need to add filter to find if there are programs [Not in the schema of this branch]
+  componentDidMount() {
+    const { database } = this.props;
+    const usesPrograms = database.objects('MasterList');
```

Comment:
Should check master lists for this store and isProgram (check out SupplierRequisitionPage.onAddMasterItems for hints)
Originally, when i was thinking about mobile logic for figuring out programs, I thought this way:
When newRequisition button is pressed, try to find all programs that have at least one order type for current store tags, then build an object with all relevant information (keeps all of the programSetting object parsing logic in one place, and only parse it once when creating requisition). If no programs with order types are found then use the previous modal, if they are found use ByProgramModal <- passing all needed information to it:
The object passed to ByProgramModal would be something like:
```
{
 programOneName: {
  orderType: { periods: [period1, period2, period3, etc..] // periods here can be realmObject
      maxMos, thresholdMos etc..
    }
   }
}
```
I haven't fully reviewed ByProgramModal, but it looks like it would be doing the queries to figure out order types etc.. No need for refactor on this, but would like to hear your thoughts

Reply:
Ah that is a bad check for programs existing, huh.

As for passing props: This is how I had it originally, passing all the program values: programs, suppliers, ordertypes and periods to `ByProgramModal`. It turned out to be a lot messier, in my opinion. Supplier requisitions page was already a sizeable component and adding the logic to manage these state transformations (which is quite a lot, if you are accounting for managing at which stage in the process you are in to reduce the amount of querying that is going on) seemed to be quite a nightmare. Although in hindsight this was also when I was trying to use the `PageInfo` prop, so that could have been a part of the problem also. 

Basically it just ended up being a bit cleaner to manage state within the modal component rather than having the modal uncontrolled and simply passing in props for values, event handlers etc. I think I would still prefer to abstract even more of the state management into more components, though

Reply:
Done here b983e71ac72281f8329e9de43dd87dbdf1ce6aac